tests/unit/test_charm_state.py

```diff
@@ -929,6 +929,41 @@ def mock_charm_state_data():
     }
 
 
+@pytest.mark.parametrize(
+    "immutable_config",
+    [
+        pytest.param("runner_storage", id="Runner storage"),
+        pytest.param("base_image", id="Base image"),
+    ],
+)
+def test_check_immutable_config_key_error(
+    mock_charm_state_path: Path,
+    mock_charm_state_data: dict[str, typing.Any],
+    immutable_config: str,
+    monkeypatch: pytest.MonkeyPatch,
+    caplog: pytest.LogCaptureFixture,
+):
+    """
+    arrange: Mock CHARM_STATE_PATH and read_text method to return previous state with same config.
```

Comment:
Shouldn't this be phrased more like "to return previous state without particular config"?

Reply:
Updated docstring to: `Mock CHARM_STATE_PATH and read_text method to return modified immutable config values` would this sound good?